openapi3/path_item.go

```diff
@@ -123,9 +123,9 @@ func (pathItem *PathItem) SetOperation(method string, operation *Operation) {
 
 // Validate returns an error if PathItem does not comply with the OpenAPI spec.
 func (pathItem *PathItem) Validate(ctx context.Context) error {
-	for _, operation := range pathItem.Operations() {
+	for method, operation := range pathItem.Operations() {
 		if err := operation.Validate(ctx); err != nil {
-			return err
+			return fmt.Errorf("path (summary: %s) failed validation for operation %s: %v", pathItem.Summary, method, err)
```

Comment:
I didn't want to modify the signature of the `Validate` method to take the `path` as input as that might mean more changes. I didn't check if it would be. This seemed like an acceptable minimal change to improve the error messaging. Let me know what you think.